src/Aspire.Hosting/ApplicationModel/InputAnnotation.cs

```diff
@@ -49,11 +46,30 @@ public InputAnnotation(string name, string? type = null, bool secret = false)
     /// </summary>
     public InputDefault? Default { get; set; }
 
-    /// <summary>
-    /// The value of the input.
-    /// </summary>
-    public string GenerateDefaultValue()
+    internal string? Value
+    {
+        get
+        {
+            if (!_hasValue)
+            {
+                _value = GenerateValue();
+                _hasValue = true;
+            }
+            return _value;
+        }
+    }
+
+    ValueTask<string?> IValueProvider.GetValueAsync(CancellationToken cancellationToken) => new(Value);
```

Comment:
Do you need this?

Reply:
I want InputAnnotation to implement IValueProvider.  That way I can remove code like:

https://github.com/dotnet/aspire/blob/12c7d857b5efcaa778a5f2a733d5e76a07e76f65/src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs#L477

Reply:
Unless you think InputAnnotation shouldn't be IValueProvider, and only InputReference should be, and it would just return `Input.Value`.

Not sure how far down IValueProvider should go.

Reply:
InputReference 👍🏾 
InputAnnotation 👎🏾